pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -459,23 +360,29 @@ type libInfo struct {
 // extractLibInfo returns the language, the image,
 // and a boolean indicating whether the library should be injected into the pod
 func (w *Webhook) extractLibInfo(pod *corev1.Pod) ([]libInfo, bool) {
-	var libInfoList []libInfo
-	var autoDetected = false
-
-	// The library version specified via annotation on the Pod takes precedence over libraries injected with Single Step Instrumentation
-	if ShouldInject(pod, w.wmeta) {
-		libInfoList = w.extractLibrariesFromAnnotations(pod)
-		if len(libInfoList) > 0 {
-			return libInfoList, autoDetected
+
+	if w.isPodEligible(pod) {
```

Comment:
We still call `isPodEligible` two times, and `isEnabledInNamespace` 4(? total). 

Mostly, we still need to disambiguate between "don't touch this pod" and "auto_instrumentation is enabled" and what we look for in terms of annotations and the auto instrumentation feature in each case. (prme-3) 

(prme-2)